python/res/job_queue/queue.py

```diff
@@ -250,6 +254,46 @@ def getJobStatus(self, job_number):
         int_status = self._get_job_status(job_number)
         return JobStatusType( int_status )
 
-
-
-
+    def is_running(self):
+        for job in self.job_list:
+            job_status = job.status
+            if (job_status ==  JobStatusType.JOB_QUEUE_PENDING or
+                job_status == JobStatusType.JOB_QUEUE_SUBMITTED or
+                job_status == JobStatusType.JOB_QUEUE_WAITING or
+                job_status == JobStatusType.JOB_QUEUE_RUNNING):
+                return True
+        return False
+
+    def fetch_next_waiting(self):
+        for job in self.job_list:
+            if job.status == JobStatusType.JOB_QUEUE_WAITING and not job.started:
+                return job
+        return None
+
+    def kill_all_jobs(self):
```

Comment:
Should toggle `user_exit` here, so that it can be communicated to `JobQueueManager.execute_queue` that it should stop submitting jobs.

Reply:
Maybe we move user_exit into python, but disjoint from the c - user_exit?

Reply:
Yeah, even better.